### PR TITLE
Admin user merchant#113

### LINKED ISSUE #113: User Story 24, Admin can make a User a Merchant
```
As an admin user
When I visit a user's profile page ("/admin/users/5")
I see a link to "upgrade" the user's account to become a merchant
When I click on that link
I am redirected to ("/admin/merchants/5") because the user is now a merchant
And I see a flash message indicating the user has been upgraded
The next time this user logs in they are now a merchant
Only admins can see the "upgrade" link
Only admins can reach any route necessary to upgrade the user to merchant status
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

### PR DESCRIPTION
**Pull request by stoic-plus**
* closes #113 

I pulled down another branch so there's extra commits. I'm cheating the contributor's metrics.
_10 points for Gryffindor!_

* Changed `Admin::MerchantsContr` toggle_role to show different flash message based on what action took place and different redirects 
* So if admin just downgraded - redirect to `admin_user_path` because merchant is now default user
  * and vice-versa

* Added button to upgrade to admin_user show page 
* Removed conditional from admin_merchant show page because it should only have option to downgrade
  * (If default user then they won't show up here - they would be in `admin_user` show)

Testing that button is not present for other users who aren't admin was unnecessary because they are already blocked from that route
* Thanks blocking and thanks to __Mexico__ for paying for that wall. You may not bring your best pero esa pared si que esta chingona

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -22,7 +22,13 @@ def toggle_role
     user = User.find(params[:id])
     user.toggle_role
     user.reload
-    redirect_to admin_merchant_path(user)
+    if user.role == "default"
+      flash[:success] = "Merchant is now a default user"
+      redirect_to admin_user_path(user)
+    else
+      flash[:success] = "User has been updgraded to Merchant"
+      redirect_to admin_merchant_path(user)
```
Comment (le3ah) [inline app/controllers/admin/merchants_controller.rb]: fancy!

File: app/views/admin/merchants/show.html.erb [hunk 2]
```diff
@@ -1,7 +1,3 @@
 Merchant Page
 
-<% if @user.role == "merchant" %>
-    <%= button_to "Downgrade to User", admin_toggle_role_path(id:params[:id]) %>
-<% elsif @user.role == 'default' %>
-    <%= button_to "Upgrade to Merchant", admin_toggle_role_path(id:params[:id]) %>
-<% end %>
\ No newline at end of file
+<%= button_to "Downgrade to User", admin_toggle_role_path(id:params[:id]) %>
```
Comment (le3ah) [inline app/views/admin/merchants/show.html.erb]: power button

File: spec/features/users/admin/admin_can_make_user_merchant_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  it 'redirects to admin_merchant and shows flash message after upgrading' do
+    visit admin_user_path(@u)
+
+    click_button "Upgrade User"
+
+    expect(current_path).to eq(admin_merchant_path(@u))
+    expect(page).to have_content("User has been updgraded to Merchant")
+  end
+
+  it 'persits user upgrade next time they login' do
```
Comment (le3ah) [inline spec/features/users/admin/admin_can_make_user_merchant_spec.rb]: great test

File: app/controllers/admin/merchants_controller.rb [hunk 4]
```diff
@@ -10,7 +10,7 @@ def show
   def toggle_status
     user = User.find(params[:user_id])
     user.switch_enabled
-    
+
     flash[:success] = user.enabled? ?
```
[the remark this answers is not in the document]
Comment (stoic-plus) [reply app/controllers/admin/merchants_controller.rb]: Ugly!?

File: spec/features/users/admin/admin_blocked_navigation_spec.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  it 'should block admin from dashboard paths' do
+    visit dashboard_path
+    expect(page).to have_content("The page you were looking for doesn't exist.")
+    expect(page.status_code).to eq(404)
+
+    visit dashboard_items_path
+    expect(page).to have_content("The page you were looking for doesn't exist.")
+    expect(page.status_code).to eq(404)
+  end
+end
```
[the remark this answers is not in the document]
Comment (stoic-plus) [reply spec/features/users/admin/admin_blocked_navigation_spec.rb]: Yes you have

File: app/controllers/admin/merchants_controller.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (tonyr729) [reply app/controllers/admin/merchants_controller.rb]: I don't think its ugly, but they do.
